grails-app/controllers/au/org/ala/spatial/portal/PortalController.groovy

```diff
@@ -98,12 +98,8 @@ class PortalController {
                     config.spApp.each { k, v ->
                         spApp.put(k, v.class.newInstance(params.get(k, v)))
                     }
-                    if (params.get("lang")) {
-                        config.i18n?.currentRegion = params.get("lang")
-                    } else {
-                        config.i18n?.currentRegion = null;
-                    }
 
+                    config.i18n?.currentRegion = org.springframework.context.i18n.LocaleContextHolder.getLocale()?.getLanguage()
```

Comment:
Take this opportunity to fix this inherited bug. Do not update `config.i18n?.currentRegion`. Instead, create a new variable and pass it to the model for use by index.gsp.